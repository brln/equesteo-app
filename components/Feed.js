import React, { Component } from 'react';
import { List, ListItem } from 'react-native-elements'
import {
  StyleSheet,
  Text,
  View
} from 'react-native';

export default class Feed extends Component {
  constructor (props) {
    super(props)
    this.state = {}
  }


  render() {
    return (
      <View>
        <List containerStyle={{marginTop: 0}}>
          {
            [...this.props.rides.map((ride, i) => (
              <ListItem
                key={i}
                title={ride.name}
                subtitle={ride.userID}
              />
            )),
            ]
          }
        </List>
      </View>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'flex-start',
    alignItems: 'stretch',
    backgroundColor: '#F5FCFF',
  },
  header: {
    padding: 20,
    fontSize: 24,
    fontWeight: 'bold'
  }
});
