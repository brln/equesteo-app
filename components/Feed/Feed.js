import React, { Component } from 'react';
import { Container, Tab, Tabs } from 'native-base';
import {
  StyleSheet,
} from 'react-native';

import RideList from './RideList'
import { RIDE } from '../../screens'
import { brand } from '../../colors'

export default class Feed extends Component {
  constructor (props) {
    super(props)
    this.showRide = this.showRide.bind(this)
    this.startRefresh = this.startRefresh.bind(this)
  }

  componentWillReceiveProps (prevProps) {
    if (prevProps.justFinishedRide) {
      this.showRide(prevProps.yourRides[0])
      this.props.justFinishedRideShown()
    }
  }

  startRefresh () {
    this.props.syncDBPull()
  }

  showRide (ride) {
    let rightButtons = []
    if (this.props.userID === ride.userID) {
      rightButtons = [
        {
          icon: require('../../img/threedot.png'),
          id: 'dropdown',
        }
      ]
    }
    this.props.navigator.push({
      screen: RIDE,
      title: ride.name,
      passProps: {
        rideID: ride._id,
      },
      navigatorButtons: {
        leftButtons: [],
        rightButtons
      },
      animationType: 'slide-up',
    });
  }

  render() {
    return (
      <Container>
        <Tabs initialPage={0} locked={true}>
          <Tab tabStyle={{backgroundColor: brand}} activeTabStyle={{backgroundColor: brand}} heading="Following">
            <RideList
              horses={this.props.horses}
              navigator={this.props.navigator}
              refreshing={this.props.refreshing}
              rides={this.props.followingRides}
              rideCarrots={this.props.rideCarrots}
              rideComments={this.props.rideComments}
              showComments={this.props.showComments}
              showRide={this.showRide}
              startRefresh={this.startRefresh}
              toggleCarrot={this.props.toggleCarrot}
              users={this.props.users}
            />
          </Tab>
          <Tab tabStyle={{backgroundColor: brand}} activeTabStyle={{backgroundColor: brand}} heading="You">
            <RideList
              horses={this.props.horses}
              navigator={this.props.navigator}
              refreshing={this.props.refreshing}
              rides={this.props.yourRides}
              rideCarrots={this.props.rideCarrots}
              rideComments={this.props.rideComments}
              showComments={this.props.showComments}
              showRide={this.showRide}
              startRefresh={this.startRefresh}
              toggleCarrot={this.props.toggleCarrot}
              users={this.props.users}
            />
          </Tab>
        </Tabs>
      </Container>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'flex-start',
    alignItems: 'stretch',
    backgroundColor: '#F5FCFF',
  },
  header: {
    padding: 20,
    fontSize: 24,
    fontWeight: 'bold'
  }
});
