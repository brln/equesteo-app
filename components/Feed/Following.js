import React, { Component } from 'react'
import { List } from 'native-base'
import {
  ScrollView,
  StyleSheet,
} from 'react-native';
import RideCard from './RideCard'


export default class Following extends Component {

  constructor (props) {
    super(props)
    this.state = {}
  }

  render() {
    return (
      <ScrollView>
        <List containerStyle={{marginTop: 0}}>
          {
            this.props.rides.map((ride, i) => (
              <RideCard
                key={i}
                ride={ride}
                showRide={this.props.showRide}
              />
            ))
          }
        </List>
      </ScrollView>
    )
  }
}

const styles = StyleSheet.create({
  rideTitle: {
    fontSize: 24
  }
});
