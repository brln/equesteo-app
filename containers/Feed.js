import memoizeOne from 'memoize-one';
import React from 'react'
import { connect } from 'react-redux'
import { Navigation } from 'react-native-navigation'

import { clearFeedMessage } from "../actions/standard";
import { syncDBPull, toggleRideCarrot } from "../actions/functional";
import BackgroundComponent from '../components/BackgroundComponent'
import { brand } from '../colors'
import Feed from '../components/Feed/Feed'
import { logRender } from '../helpers'
import {
  FIND_PEOPLE,
  FIRST_START,
  HORSE_PROFILE,
  PROFILE,
  RIDE
} from '../screens'

class FeedContainer extends BackgroundComponent {
   static options() {
    return {
      topBar: {
        title: {
          text: "Feed",
          color: 'white',
          fontSize: 20
        },
        leftButtons: [{
          id: 'sideMenu',
          icon: require('../img/hamburger.png'),
          color: 'white'
        }],
        rightButtons: [{
          id: 'findFriends',
          icon: require('../img/findPeople.png'),
          color: 'white'
        }],
        background: {
          color: brand,
        },
        elevation: 0
      },
      layout: {
        orientation: ['portrait']
      }
    };
  }

  constructor (props) {
    super(props)
    this.state = {
      refreshing: false,
      lastFullSync: null,
      firstStartPopped: false,
      ridePopped: false,
    }
    this.toggleCarrot = this.toggleCarrot.bind(this)
    this.showProfile = this.showProfile.bind(this)
    this.showRide = this.showRide.bind(this)
    this.showHorseProfile = this.showHorseProfile.bind(this)
    this.syncDBPull = this.syncDBPull.bind(this)

    this.followIDs = this.followIDs.bind(this)
    this.followingRides = this.followingRides.bind(this)
    this.horseOwnerIDs = this.horseOwnerIDs.bind(this)
    this.filteredHorses = this.filteredHorses.bind(this)
    this.filteredHorseUsers = this.filteredHorseUsers.bind(this)
    this.yourRides = this.yourRides.bind(this)

    this.memoizeFollowIDs = memoizeOne(this.followIDs)
    this.memoizeFollowingRides = memoizeOne(this.followingRides)
    this.memoizeHorseOwnerIDs = memoizeOne(this.horseOwnerIDs)
    this.memoizeFilteredHorses = memoizeOne(this.filteredHorses)
    this.memoizeFilteredHorseUsers = memoizeOne(this.filteredHorseUsers)
    this.memoizeRideHorses = memoizeOne(this.rideHorses.bind(this))
    this.memoizeYourRides = memoizeOne(this.yourRides)

    Navigation.events().bindComponent(this)
    this.navigationButtonPressed = this.navigationButtonPressed.bind(this)
  }

  navigationButtonPressed({ buttonId }) {
    if (buttonId === 'sideMenu') {
      Navigation.mergeOptions(this.props.componentId, {
        sideMenu: {
          left: {
            visible: true,
          }
        }
      });
    } else if (buttonId === 'findFriends') {
      Navigation.push(this.props.componentId, {
        component: {
          name: FIND_PEOPLE,
        }
      })
    }
  }

  static getDerivedStateFromProps (nextProps, prevState) {
    const nextState = {}
    nextState.lastFullSync = nextProps.lastFullSync
    if (prevState.lastFullSync !== nextProps.lastFullSync || nextProps.fullSyncFail) {
      nextState.refreshing = false
    }
    return nextState
  }

  componentDidUpdate () {
    if (this.props.popShowRideNow && !this.props.awaitingFullSync) {
      const showRide = this.props.rides.get(this.props.popShowRide.get('rideID'))
      if (showRide && !this.state.ridePopped) {
        // PushNotification.showNotification gets called when it shouldn't if the app reboots unexpectedly.
        // This makes popShowRideNow get set when it shouldn't, so make sure we have a ride to show here.
        this.showRide(showRide, this.props.popShowRide.get('scrollToComments'), true)
        this.setState({
          ridePopped: true
        })
      }
    } else if (this.state.ridePopped) {
      this.setState({
        ridePopped: false
      })
    }

    if (this.props.user && (!this.state.firstStartPopped && !this.props.user.get('finishedFirstStart'))) {
      this.showFirstStart()
      this.setState({
        firstStartPopped: true
      })
    }
  }

  showHorseProfile (horse, ownerID) {
    Navigation.push(this.props.componentId, {
      component: {
        name: HORSE_PROFILE,
        title: horse.get('name'),
        passProps: {
          horse,
          ownerID
        }
      }
    })
  }

  showFirstStart () {
     Navigation.push(this.props.componentId, {
       component: {
         name: FIRST_START,
       }
     })
  }

  showProfile (user) {
    Navigation.push(this.props.componentId, {
      component: {
        name: PROFILE,
        passProps: {
          profileUser: user,
        }
      }
    })
  }

  showRide (ride, skipToComments, isPopShow) {
    Navigation.push(this.props.componentId, {
      component: {
        name: RIDE,
        passProps: {
          rideID: ride.get('_id'),
          skipToComments,
          isPopShow,
        }
      }
    });
  }

  syncDBPull () {
    this.setState({
      refreshing: true
    })
    this.props.dispatch(syncDBPull())
  }

  toggleCarrot (rideID) {
    this.props.dispatch(toggleRideCarrot(rideID))
  }

  yourRides (rides, userID) {
    return rides.valueSeq().filter(
      (r) => r.get('userID') === userID && r.get('deleted') !== true
    ).sort((a, b) =>
      b.get('startTime') - a.get('startTime')
    ).toList()
  }

  followIDs (follows, userID) {
    return follows.valueSeq().filter(
      f => f.get('deleted') !== true && f.get('followerID') === userID
    ).map(
      f => f.get('followingID')
    ).toList()
  }

  followingRides (follows, userID, rides) {
    return rides.valueSeq().filter(
      r => r.get('isPublic') === true // is a public ride
        && r.get('deleted') !== true // hasn't been deleted
        && (r.get('userID') === userID || this.memoizeFollowIDs(follows, userID).indexOf(r.get('userID')) >= 0) // user hasn't removed follow
    ).sort(
      (a, b) => b.get('startTime') - a.get('startTime')
    ).toList()
  }

  filteredHorses (follows, userID, horseUsers, horses) {
    return horseUsers.filter(h => {
      return this.memoizeFollowIDs(follows, userID).indexOf(h.get('userID')) >= 0 || h.get('userID') === userID
    }).mapEntries(([horseUserID, horseUser]) => {
      return [horseUser.get('horseID'), horses.get(horseUser.get('horseID'))]
    })
  }

  filteredHorseUsers (follows, userID, horseUsers) {
    return horseUsers.filter(h => {
      return this.memoizeFollowIDs(follows, userID).indexOf(h.get('userID')) >= 0 || h.get('userID') === userID
    })
  }


  horseOwnerIDs (horseUsers) {
    return horseUsers.filter(hu => {
      return hu.get('owner') === true
    }).mapEntries(([horseUserID, horseUser]) => {
      return [horseUser.get('horseID'), horseUser.get('userID')]
    })
  }

  rideHorses (rideHorses) {
    return rideHorses.filter(rh => {
      return rh.get('deleted') !== true
    })
  }

  render() {
    logRender('feedContainer')
    return (
      <Feed
        deleteRide={this.deleteRide}
        feedMessage={this.props.feedMessage}
        followingRides={this.memoizeFollowingRides(this.props.follows, this.props.userID, this.props.rides)}
        horses={this.memoizeFilteredHorses(this.props.follows, this.props.userID, this.props.horseUsers, this.props.horses)}
        horsePhotos={this.props.horsePhotos}
        horseOwnerIDs={this.memoizeHorseOwnerIDs(this.props.horseUsers)}
        horseUsers={this.memoizeFilteredHorseUsers(this.props.follows, this.props.userID, this.props.horseUsers)}
        rideHorses={this.memoizeRideHorses(this.props.rideHorses)}
        refreshing={this.state.refreshing}
        rideCarrots={this.props.rideCarrots.toList()}
        rideComments={this.props.rideComments.toList()}
        ridePhotos={this.props.ridePhotos}
        showHorseProfile={this.showHorseProfile}
        showComments={this.showComments}
        showProfile={this.showProfile}
        showRide={this.showRide}
        syncDBPull={this.syncDBPull}
        toggleCarrot={this.toggleCarrot}
        userID={this.props.userID}
        users={this.props.users}
        userPhotos={this.props.userPhotos}
        yourRides={this.memoizeYourRides(this.props.rides, this.props.userID)}
      />
    )
  }
}

function mapStateToProps (state) {
  const pouchState = state.get('pouchRecords')
  const localState = state.get('localState')
  const userID = localState.get('userID')
  return {
    activeComponent: localState.get('activeComponent'),
    awaitingFullSync: localState.get('awaitingFullSync'),
    feedMessage: localState.get('feedMessage'),
    follows: pouchState.get('follows'),
    fullSyncFail: localState.get('fullSyncFail'),
    horses: pouchState.get('horses'),
    horsePhotos: pouchState.get('horsePhotos'),
    horseUsers: pouchState.get('horseUsers'),
    justFinishedRide: localState.get('justFinishedRide'),
    lastFullSync: localState.get('lastFullSync'),
    popShowRide: localState.get('popShowRide'),
    popShowRideNow: localState.get('popShowRideNow'),
    rides: pouchState.get('rides'),
    rideCarrots: pouchState.get('rideCarrots'),
    rideComments: pouchState.get('rideComments'),
    rideHorses: pouchState.get('rideHorses'),
    ridePhotos: pouchState.get('ridePhotos'),
    users: pouchState.get('users'),
    userID,
    user: pouchState.getIn(['users', localState.get('userID')]),
    userPhotos: pouchState.get('userPhotos')
  }
}

export default connect(mapStateToProps)(FeedContainer)
