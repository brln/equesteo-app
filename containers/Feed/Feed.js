import memoizeOne from 'memoize-one'
import React from 'react'
import { connect } from 'react-redux'
import { Navigation } from 'react-native-navigation'

import { pulldownSync, toggleRideCarrot } from "../../actions/functional"
import BackgroundComponent from '../../components/BackgroundComponent'
import { brand } from '../../colors'
import Feed from '../../components/Feed/Feed'
import { logRender } from '../../helpers'
import {
  FIRST_START,
  HORSE_PROFILE,
  PROFILE,
  RECORDER,
  RIDE,
  RIDE_BUTTON,
} from '../../screens'

class FeedContainer extends BackgroundComponent {
   static options() {
    return {
      topBar: {
        title: {
          text: "Feed",
          color: 'white',
          fontSize: 20
        },
        leftButtons: [{
          id: 'sideMenu',
          icon: require('../../img/hamburger.png'),
          color: 'white'
        }],
        background: {
          color: brand,
        },
        elevation: 0
      },
      layout: {
        orientation: ['portrait']
      }
    };
  }

  constructor (props) {
    super(props)
    this.state = {
      refreshing: false,
      lastFullSync: null,
      firstStartPopped: false,
      ridePopped: false,
      recorderDebounce: false,
    }

    this.followIDs = this.followIDs.bind(this)
    this.followingRides = this.followingRides.bind(this)
    this.horseOwnerIDs = this.horseOwnerIDs.bind(this)
    this.filteredHorses = this.filteredHorses.bind(this)
    this.filteredHorseUsers = this.filteredHorseUsers.bind(this)
    this.makeSureDrawerClosed = this.makeSureDrawerClosed.bind(this)
    this.showProfile = this.showProfile.bind(this)
    this.showRide = this.showRide.bind(this)
    this.showHorseProfile = this.showHorseProfile.bind(this)
    this.syncDB = this.syncDB.bind(this)
    this.toggleCarrot = this.toggleCarrot.bind(this)
    this.openRecorder = this.openRecorder.bind(this)
    this.yourRides = this.yourRides.bind(this)

    this.memoizeFollowIDs = memoizeOne(this.followIDs)
    this.memoizeFollowingRides = memoizeOne(this.followingRides)
    this.memoizeHorseOwnerIDs = memoizeOne(this.horseOwnerIDs)
    this.memoizeFilteredHorses = memoizeOne(this.filteredHorses)
    this.memoizeFilteredHorseUsers = memoizeOne(this.filteredHorseUsers)
    this.memoizeRideHorses = memoizeOne(this.rideHorses.bind(this))
    this.memoizeYourRides = memoizeOne(this.yourRides)

    Navigation.events().bindComponent(this)
    this.navigationButtonPressed = this.navigationButtonPressed.bind(this)
    Navigation.mergeOptions(this.props.componentId, {
      topBar: {
        rightButtons: [{
          component: {
            id: RIDE_BUTTON,
            name: RIDE_BUTTON,
            passProps: {
              onPress: this.openRecorder,
            }
          }
        }],
      }
    })
  }

  openRecorder () {
    if (!this.state.recorderDebounce) {
      this.setState({
        recorderDebounce: true
      })
      Navigation.push(this.props.activeComponent, {
        component: {
          name: RECORDER,
          id: RECORDER
        }
      }).then(() => {
        Navigation.mergeOptions(this.props.componentId, {
          sideMenu: {
            left: {
              visible: false,
            }
          }
        })
        this.setState({
          recorderDebounce: false
        })
      })
    }
  }

  navigationButtonPressed({ buttonId }) {
    if (buttonId === 'sideMenu') {
      Navigation.mergeOptions(this.props.componentId, {
        sideMenu: {
          left: {
            visible: true,
          }
        }
      });
    }
  }

  static getDerivedStateFromProps (nextProps, prevState) {
    const nextState = {}
    nextState.lastFullSync = nextProps.lastFullSync
    if (prevState.lastFullSync !== nextProps.lastFullSync || nextProps.fullSyncFail) {
      nextState.refreshing = false
    }
    return nextState
  }

  componentDidUpdate () {
    if (this.props.popShowRideNow && this.props.popShowRide && this.props.popShowRide.get('rideID')) {
      const showRide = this.props.rides.get(this.props.popShowRide.get('rideID'))
      if (showRide && !this.state.ridePopped) {
        // PushNotification.showNotification gets called when it shouldn't if the app reboots unexpectedly.
        // This makes popShowRideNow get set when it shouldn't, so make sure we have a ride to show here.
        this.showRide(showRide, this.props.popShowRide.get('scrollToComments'), true)
        this.setState({
          ridePopped: true
        })
      }
    } else if (this.state.ridePopped) {
      this.setState({
        ridePopped: false
      })
    }

    if (this.props.user && (!this.state.firstStartPopped && !this.props.user.get('finishedFirstStart'))) {
      this.showFirstStart()
      this.setState({
        firstStartPopped: true
      })
    }
  }

  makeSureDrawerClosed () {
    Navigation.mergeOptions(this.props.componentId, {
      sideMenu: {
        left: {
          visible: false,
        }
      }
    })
  }

  showHorseProfile (horse, ownerID) {
    Navigation.push(this.props.componentId, {
      component: {
        name: HORSE_PROFILE,
        title: horse.get('name'),
        passProps: {
          horse,
          ownerID
        }
      }
    }).then(() => {
      this.makeSureDrawerClosed()
    })
  }

  showFirstStart () {
     Navigation.push(this.props.componentId, {
       component: {
         name: FIRST_START,
       }
     })
  }

  showProfile (user) {
    Navigation.push(this.props.componentId, {
      component: {
        name: PROFILE,
        passProps: {
          profileUser: user,
        }
      }
    }).then(() => {
      this.makeSureDrawerClosed()
    })
  }

  showRide (ride, skipToComments, isPopShow) {
    Navigation.push(this.props.componentId, {
      component: {
        name: RIDE,
        passProps: {
          rideID: ride.get('_id'),
          skipToComments,
          isPopShow,
        }
      }
    }).then(() => {
      this.makeSureDrawerClosed()
    })
  }

  syncDB () {
    this.setState({
      refreshing: true
    })
    this.props.dispatch(pulldownSync())
  }

  toggleCarrot (rideID) {
    this.props.dispatch(toggleRideCarrot(rideID))
  }

  yourRides (rides, userID) {
    return rides.valueSeq().filter(
      (r) => r.get('userID') === userID && r.get('deleted') !== true
    ).sort((a, b) =>
      b.get('startTime') - a.get('startTime')
    ).toList().slice(0, 30)
  }

  followIDs (follows, userID) {
    return follows.valueSeq().filter(
      f => f.get('deleted') !== true && f.get('followerID') === userID
    ).map(
      f => f.get('followingID')
    ).toList()
  }

  followingRides (follows, userID, rides) {
    return rides.valueSeq().filter(
      r => r.get('isPublic') === true // is a public ride
        && r.get('deleted') !== true // hasn't been deleted
        && (r.get('userID') === userID || this.memoizeFollowIDs(follows, userID).indexOf(r.get('userID')) >= 0) // user hasn't removed follow
    ).sort(
      (a, b) => b.get('startTime') - a.get('startTime')
    ).toList().slice(0, 50)
  }

  filteredHorses (follows, userID, horseUsers, horses) {
    return horseUsers.filter(h => {
      return this.memoizeFollowIDs(follows, userID).indexOf(h.get('userID')) >= 0 || h.get('userID') === userID
    }).mapEntries(([horseUserID, horseUser]) => {
      return [horseUser.get('horseID'), horses.get(horseUser.get('horseID'))]
    })
  }

  filteredHorseUsers (follows, userID, horseUsers) {
    return horseUsers.filter(h => {
      return this.memoizeFollowIDs(follows, userID).indexOf(h.get('userID')) >= 0 || h.get('userID') === userID
    })
  }


  horseOwnerIDs (horseUsers) {
    return horseUsers.filter(hu => {
      return hu.get('owner') === true
    }).mapEntries(([horseUserID, horseUser]) => {
      return [horseUser.get('horseID'), horseUser.get('userID')]
    })
  }

  rideHorses (rideHorses) {
    return rideHorses.filter(rh => {
      return rh.get('deleted') !== true
    })
  }

  render() {
    logRender('feedContainer')
    return (
      <Feed
        deleteRide={this.deleteRide}
        feedMessage={this.props.feedMessage}
        followingRides={this.memoizeFollowingRides(this.props.follows, this.props.userID, this.props.rides)}
        horses={this.memoizeFilteredHorses(this.props.follows, this.props.userID, this.props.horseUsers, this.props.horses)}
        horsePhotos={this.props.horsePhotos}
        horseOwnerIDs={this.memoizeHorseOwnerIDs(this.props.horseUsers)}
        horseUsers={this.memoizeFilteredHorseUsers(this.props.follows, this.props.userID, this.props.horseUsers)}
        rideHorses={this.memoizeRideHorses(this.props.rideHorses)}
        refreshing={this.state.refreshing}
        rideCarrots={this.props.rideCarrots.toList()}
        rideComments={this.props.rideComments.toList()}
        ridePhotos={this.props.ridePhotos}
        showHorseProfile={this.showHorseProfile}
        showComments={this.showComments}
        showProfile={this.showProfile}
        showRide={this.showRide}
        syncDB={this.syncDB}
        toggleCarrot={this.toggleCarrot}
        userID={this.props.userID}
        users={this.props.users}
        userPhotos={this.props.userPhotos}
        yourRides={this.memoizeYourRides(this.props.rides, this.props.userID)}
      />
    )
  }
}

function mapStateToProps (state) {
  const localState = state.get('localState')
  const pouchState = state.get('pouchRecords')
  const userID = localState.get('userID')
  return {
    activeComponent: localState.get('activeComponent'),
    feedMessage: localState.get('feedMessage'),
    follows: pouchState.get('follows'),
    fullSyncFail: localState.get('fullSyncFail'),
    horses: pouchState.get('horses'),
    horsePhotos: pouchState.get('horsePhotos'),
    horseUsers: pouchState.get('horseUsers'),
    justFinishedRide: localState.get('justFinishedRide'),
    lastFullSync: localState.get('lastFullSync'),
    popShowRide: localState.get('popShowRide'),
    popShowRideNow: localState.get('popShowRideNow'),
    rides: pouchState.get('rides'),
    rideCarrots: pouchState.get('rideCarrots'),
    rideComments: pouchState.get('rideComments'),
    rideHorses: pouchState.get('rideHorses'),
    ridePhotos: pouchState.get('ridePhotos'),
    users: pouchState.get('users'),
    userID,
    user: pouchState.getIn(['users', localState.get('userID')]),
    userPhotos: pouchState.get('userPhotos')
  }
}

export default connect(mapStateToProps)(FeedContainer)
