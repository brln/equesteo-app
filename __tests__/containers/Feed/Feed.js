import configureStore from 'redux-mock-store'
import React from "react"
import {fromJS, Map} from "immutable"
import renderer from "react-test-renderer"
import { Provider } from 'react-redux'
import thunk from "redux-thunk"

import Feed from '../../../containers/Feed/Feed'


const mockStore = configureStore([thunk])

test('render', () => {
  const store = mockStore(fromJS({
    currentRide: {},
    pouchRecords: {
      follows: {},
      horseUsers: {},
      notifications: {},
      rideHorses: {},
      rideCarrots: {},
      rideComments: {},
      rides: {},
      users: {},
    },
    localState: {}
  }))

  const tree = renderer.create(
    <Provider store={store}>
      <Feed />
    </Provider>)
  expect(tree.toJSON()).toMatchSnapshot()
})

