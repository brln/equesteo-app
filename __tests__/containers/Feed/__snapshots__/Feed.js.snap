// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
      "height": 1334,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        Object {
          "flex": 10,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "borderBottomWidth": 0.5,
            "borderColor": "#a7a6ab",
            "elevation": 3,
            "flexDirection": "row",
            "height": 50,
            "justifyContent": "space-around",
            "shadowColor": undefined,
            "shadowOffset": undefined,
            "shadowOpacity": undefined,
            "shadowRadius": undefined,
          },
          Object {
            "backgroundColor": "#F8F8F8",
          },
          Object {},
        ]
      }
    >
      <View
        accessible={true}
        clickable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "flex": 1,
            "opacity": 1,
          }
        }
      >
        <View
          active={true}
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#F8F8F8",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "center",
              },
              Object {
                "backgroundColor": "#5bc0eb",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007aff",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": "600",
                  "marginLeft": 7,
                  "marginRight": 7,
                },
                Object {
                  "fontSize": 15,
                },
                Object {
                  "color": "white",
                },
                Object {
                  "color": "white",
                },
              ]
            }
            uppercase={false}
          >
            Following
          </Text>
        </View>
      </View>
      <View
        accessible={true}
        clickable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "flex": 1,
            "opacity": 1,
          }
        }
      >
        <View
          active={false}
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#F8F8F8",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "center",
              },
              Object {
                "backgroundColor": "#5bc0eb",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#6b6b6b",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "marginLeft": 7,
                  "marginRight": 7,
                },
                Object {
                  "fontSize": 15,
                },
                undefined,
                Object {
                  "color": "#6b6b6b",
                },
              ]
            }
            uppercase={false}
          >
            You
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "white",
            "bottom": 0,
            "height": 4,
            "left": 0,
            "position": "absolute",
            "width": 375,
          }
        }
      />
    </View>
    <RCTScrollView
      alwaysBounceVertical={false}
      automaticallyAdjustContentInsets={false}
      contentOffset={
        Object {
          "x": 0,
        }
      }
      directionalLockEnabled={true}
      horizontal={true}
      keyboardDismissMode="on-drag"
      keyboardShouldPersistTaps="handled"
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      pagingEnabled={true}
      scrollEnabled={false}
      scrollEventThrottle={16}
      scrollsToTop={false}
      showsHorizontalScrollIndicator={false}
    >
      <View>
        <View
          style={
            Object {
              "width": 750,
            }
          }
        >
          <View
            activeTabStyle={
              Object {
                "backgroundColor": "#5bc0eb",
              }
            }
            activeTextStyle={
              Object {
                "color": "white",
              }
            }
            heading="Following"
            style={
              Object {
                "backgroundColor": "#FFF",
                "flex": 1,
              }
            }
            tabStyle={
              Object {
                "backgroundColor": "#5bc0eb",
              }
            }
          >
            <RCTScrollView
              containerStyle={
                Object {
                  "marginTop": 0,
                }
              }
              data={Array []}
              disableVirtualization={false}
              getItem={[Function]}
              getItemCount={[Function]}
              horizontal={false}
              initialNumToRender={3}
              keyExtractor={[Function]}
              maxToRenderPerBatch={2}
              onContentSizeChange={[Function]}
              onEndReachedThreshold={2}
              onLayout={[Function]}
              onMomentumScrollEnd={[Function]}
              onRefresh={[Function]}
              onScroll={[Function]}
              onScrollBeginDrag={[Function]}
              onScrollEndDrag={[Function]}
              refreshControl={
                <RefreshControlMock
                  onRefresh={[Function]}
                  refreshing={false}
                />
              }
              refreshing={false}
              renderItem={[Function]}
              renderSectionHeader={[Function]}
              scrollEventThrottle={50}
              sections={Array []}
              stickyHeaderIndices={Array []}
              stickySectionHeadersEnabled={false}
              updateCellsBatchingPeriod={50}
              windowSize={21}
            >
              <RCTRefreshControl />
              <View />
            </RCTScrollView>
          </View>
        </View>
        <View
          style={
            Object {
              "width": 750,
            }
          }
        >
          <View
            heading="You"
          />
        </View>
      </View>
    </RCTScrollView>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#5bc0eb",
        "borderTopColor": "#e8eae8",
        "borderTopWidth": 1,
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "center",
        "paddingBottom": 5,
      }
    }
  >
    <View
      accessible={true}
      clickable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "opacity": 1,
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "height": 44.46666666666667,
            "width": 44.46666666666667,
          }
        }
      />
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 10,
            "textAlign": "center",
          }
        }
      >
        Go Ride!
      </Text>
    </View>
    <View
      accessible={true}
      clickable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "opacity": 1,
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "height": 44.46666666666667,
            "width": 44.46666666666667,
          }
        }
      />
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 10,
            "textAlign": "center",
          }
        }
      >
        Training
      </Text>
    </View>
    <View
      accessible={true}
      clickable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "opacity": 1,
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "height": 44.46666666666667,
            "width": 44.46666666666667,
          }
        }
      />
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 10,
            "textAlign": "center",
          }
        }
      >
        Leaderboards
      </Text>
    </View>
    <View
      accessible={true}
      clickable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "opacity": 1,
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "height": 44.46666666666667,
            "width": 44.46666666666667,
          }
        }
      />
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 10,
            "textAlign": "center",
          }
        }
      >
        More
      </Text>
    </View>
  </View>
</View>
`;
